tests/unit/_patched_charm.py

```diff
@@ -34,7 +34,7 @@ class _PGSQLPatch:
     def __init__(self):
         # borrow some code from
         # https://github.com/canonical/ops-lib-pgsql/blob/master/tests/test_client.py
-        self._leadership_data = {}
+        self._leadership_data = {}  # type: ignore
```

Comment:
```suggestion
        self._leadership_data:dict = {}
```
would this work here?

Reply:
Not tested, but I think it will need both the key type and value type.

Reply:
I'm guessing mypy would not complain once the dict type is used

Reply:
Yes, but unfortunately, this is not the case. This is used internally in `pgsql` charm library.